AArch64: Initial version of ARM64BinaryEncoding.cpp

knn-k opened the pull request:
This commit implements the initial version of ARM64BinaryEncoding.cpp
for instruction classes defined in ARM64Instruction.hpp.

Signed-off-by: knn-k <konno@jp.ibm.com>

0xdaryl commented on compiler/aarch64/codegen/ARM64BinaryEncoding.cpp:
```diff
@@ -0,0 +1,235 @@
+int32_t OMR::ARM64::Instruction::estimateBinaryLength(int32_t currentEstimate)
+   {
+   self()->setEstimatedBinaryLength(ARM64_INSTRUCTION_LENGTH);
+   return currentEstimate + self()->getEstimatedBinaryLength();
```
This part of the instruction hierarchy isn't extensible, so `self()` usage in this function isn't strictly correct.  Please remove.

knn-k commented:
I updated `ARM64BinaryEncoding.cpp` reflecting the review comments.
I also removed `OMRInstOpCode.hpp` from the commit because the change has been already merged by #2762.

knn-k commented:
Travis CI build has failed due to an `apt-get` error.

0xdaryl commented:
The restarted CI build passed.  Merging.

0xdaryl commented:
This is an AArch64-specific change.  A genie build is not required.
